pennylane_lightning/core/bindings/Bindings_nb.hpp

```diff
@@ -589,26 +608,43 @@ void registerBackendAgnosticMeasurements(PyClass &pyclass) {
 }
 
 /**
- * @brief Register the adjoint Jacobian method.
+ * @brief Register AdjointJacobian class.
+ *
+ * @tparam StateVectorT State vector class.
+ * @param m Nanobind module.
  */
-template <class StateVectorT>
-auto registerAdjointJacobian(
-    AdjointJacobian<StateVectorT> &adjoint_jacobian, const StateVectorT &sv,
-    const std::vector<std::shared_ptr<Observable<StateVectorT>>> &observables,
-    const OpsData<StateVectorT> &operations,
-    const std::vector<std::size_t> &trainableParams)
-    -> nb::ndarray<typename StateVectorT::PrecisionT, nb::numpy, nb::c_contig> {
+template <class StateVectorT> void registerAdjointJacobian(nb::module_ &m) {
     using PrecisionT = typename StateVectorT::PrecisionT;
-    std::vector<PrecisionT> jac(observables.size() * trainableParams.size(),
-                                PrecisionT{0.0});
-    const JacobianData<StateVectorT> jd{operations.getTotalNumParams(),
-                                        sv.getLength(),
-                                        sv.getData(),
-                                        observables,
-                                        operations,
-                                        trainableParams};
-    adjoint_jacobian.adjointJacobian(std::span{jac}, jd, sv);
-    return createNumpyArrayFromVector<PrecisionT>(jac);
+
+    const std::string bitsize =
+        std::to_string(sizeof(std::complex<PrecisionT>) * 8);
+
+    std::string class_name = "AdjointJacobianC" + bitsize;
```

Comment:
wouldn't be cheaper to check the `PrecisionT` type and set `bitsize_str` either `64` or `128`?